middleware/auth/httpauth_psk.go

```diff
@@ -27,22 +28,42 @@ func NewPSK(key []byte, issuer string) (*PSK, error) {
 
 // Check implements AuthCheck
 func (p *PSK) Check(_ context.Context, r *http.Request) bool {
+	ctx := r.Context()
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "middleware/auth/PSK.Check").
+		Logger()
+	ctx = log.WithContext(ctx)
+
 	wt, ok := fromHeader(r)
 	if !ok {
+		log.Debug().Msg("failed to retrieve jwt from header")
 		return false
 	}
 	tok, err := jwt.ParseSigned(wt)
 	if err != nil {
+		log.Debug().Err(err).Msg("failed to parse jwt")
 		return false
 	}
 	cl := jwt.Claims{}
 	if err := tok.Claims(p.key, &cl); err != nil {
+		log.Debug().Err(err).Msg("failed to parse jwt")
 		return false
 	}
+
+	for i, iss := range p.iss {
+		if iss == cl.Issuer {
+			break
+		}
+		if i == len(p.iss)-1 {
+			log.Debug().Err(err).Str("iss", cl.Issuer).Msg("could not verify issuer")
+			return false
+		}
+	}
```

Comment:
This should go after the Validate call. As a general security principal, the payload shouldn't be used before it's trusted (in this case, validated).

I also think this should fail-closed, e.g.
```suggestion
	ok := false
	for _, iss := range p.iss {
		if iss == cl.Issuer {
			ok = true
			break
		}
	}
	if !ok {
		log.Debug().Err(err).Str("iss", cl.Issuer).Msg("could not verify issuer")
		return false
	}
```